_posts/2024-07-08-async-embedded-engine.adoc

```diff
@@ -0,0 +1,208 @@
+---
+layout: post
+title:  Debezium asynchronous engine
+date:   2024-07-08 00:00:00 -0000
+tags: [ machine-learning, flink, spark, online-learning, examples, apache-kafka ]
+featured: true
+author: vjuranek
+---
+
+Debezium provides a way to run the connectors directly within Debezium from the very beginning of the project.
+The way how it was provided has changed over the time and it still evolves.
+This article will describe another evolution step in this regard -  new implementation of Debezium engine.
+
++++<!-- more -->+++
+
+== A bit of history
+
+The capability to run the connector directly within Debezium was not present from the first Debezium commit, but was https://github.com/debezium/debezium/commit/2da5b37f767393847f2ee25a1465e501b5633473[added] really very early in the Debezium development as part of https://issues.redhat.com/browse/DBZ-1[DBZ-1].
+This `EmbeddedEngine` was used mostly for testing.
+However, over the time it has evolved into a full-fledged runtime platform for the connectors, with the support for storing offsets, schema history etc.
+Later on, a public API for `DebeziumEngine` was defined ( https://issues.redhat.com/browse/DBZ-234[DBZ-234]).
+The interface decoupled the user-facing API from the implementation and provided the ability to replace the implementation by a different one.
+Shortly after introducing the Debezium engine API, a new Debezium server was created as part of https://issues.redhat.com/browse/DBZ-651[DBZ-651].
+While `EmbeddedEngine` had to be wrapped into another application which would consume the records, the Debezium server provided a way to run Debezium outside the Kafka Connect cluster without such a wrapper.
+The Debezium server provided several sinks, so the users could use the server out-of-the-box without any further coding, but the Debezium server is still powered by the Debezium engine.
+As the popularity of the Debezium server increased, more and more sinks were added.
+This resulted in the separating Debezium server into https://github.com/debezium/debezium-server/[separate GitHub project].
+The separation of the Debezium server repository happened relatively recently ( https://issues.redhat.com/browse/DBZ-6049[DBZ-6049]).
```

Comment:
I wonder as a part of this history lesson, would it make sense to also include the evolution of the Operator here just as a tidbit to let users be aware that Debezium Server is the runtime under the K8 operator too?  If so, then I think this gives users 3 options, where two slightly differ only between standalone vs K8s.

Reply:
@Naros thanks, good idea! Will add it.

Reply:
done